osier/equations.py

```diff
@@ -118,14 +120,18 @@ def objective_from_energy(technology_list, attribute, solved_dispatch_model):
     non_dispatch_techs = get_nondispatchable_techs(technology_list)
     column_names = get_tech_names(technology_list)
     dispatch_results = solved_dispatch_model.results
+    power_units = solved_dispatch_model.power_units
+    time_delta = solved_dispatch_model.time_delta
+    dispatch_values = dispatch_results[column_names].values.T*power_units*time_delta
 
-    attributes = np.array([getattr(t, attribute)
+
+    attributes = unyt_array([getattr(t, attribute)
                            for t in dispatch_techs + non_dispatch_techs
                            if hasattr(t, attribute)])
 
     objective_value = np.dot(
         attributes,
-        dispatch_results[column_names].values.T).sum()
+        dispatch_values).sum()
```

Comment:
Previously didn't multiply by `time_delta` or `power_units`, just making sure that is intended.

Reply:
Yes it is intended, just makes it more robust.